pkg/network/cni.go

```diff
@@ -0,0 +1,61 @@
+package network
+
+import (
+	"encoding/json"
+	"fmt"
+	"strings"
+
+	"github.com/pkg/errors"
+
+	operv1 "github.com/openshift/api/operator/v1"
+)
+
+const CNINetworkName = "openshift"
+
+// Whatever network is the default network should use this as the target CNI
+// configuration file.
+const CNIConfigPath = "/etc/cni/net.d/80-openshift.conflist"
+
+// validateChainedPlugins ensures that all supplied chained plugins are some
+// kind of reasonable cni configuration
+func validateChainedPlugins(conf *operv1.NetworkSpec) []error {
+	out := []error{}
+
+	if len(conf.DefaultNetwork.ChainedPlugins) > 0 && conf.DefaultNetwork.Type != operv1.NetworkTypeOpenShiftSDN {
+		out = append(out, errors.Errorf("network type %s does not support chained plugins", conf.DefaultNetwork.Type))
+	}
+
+	for i, entry := range conf.DefaultNetwork.ChainedPlugins {
+		var m map[string]interface{}
+		if err := json.Unmarshal([]byte(entry), &m); err != nil {
+			out = append(out, errors.Wrapf(err, "invalid json in Spec.DefaultNetwork.ChainedPlugin[%d]", i))
+			continue
+		}
+
+		_, ok := m["type"]
+		if !ok {
+			out = append(out, errors.Errorf("invalid CNI plugin entry in Spec.DefaultNetwork.ChainedPlugin[%d]: must have 'type' key", i))
+		}
+	}
+
+	return out
+}
+
+// makeCNIConfig merges the "primary" plugin config with any desired
```

Comment:
perhaps say it merges the primary "plugin config JSON" with any chained plugin JSON config? eg that these random strings are actually JSON.